src/LinAlg/hiopVectorPar.cpp

```diff
@@ -153,25 +153,26 @@ void hiopVectorPar::copyFromStarting(int start_index/*_in_src*/,const hiopVector
   memcpy(data_+start_index, v.data_, v.n_local_*sizeof(double));
 }
 
-void hiopVectorPar::startingAtCopyFromStartingAt(int start_idx_src, 
-						 const hiopVector& v_, 
-						 int start_idx_dest)
+void hiopVectorPar::startingAtCopyFromStartingAt(int start_idx_dest, 
+						 const hiopVector& v_in, 
+						 int start_idx_src)
 {
 #ifdef HIOP_DEEPCHECKS
   assert(n_local_==n_ && "only for local/non-distributed vectors");
 #endif
-  assert((start_idx_src>=0 && start_idx_src<this->n_local_) || this->n_local_==0);
-  const hiopVectorPar& v = dynamic_cast<const hiopVectorPar&>(v_);
-  assert((start_idx_dest>=0 && start_idx_dest<v.n_local_) || v.n_local_==0);
+  assert((start_idx_dest>=0 && start_idx_dest<this->n_local_) || this->n_local_==0);
+  const hiopVectorPar& v = dynamic_cast<const hiopVectorPar&>(v_in);
+  assert((start_idx_src>=0 && start_idx_src<v.n_local_) || v.n_local_==0);
+
+  int howManyToCopy = this->n_local_ - start_idx_dest;
+  const int howManyToCopySrc = v.n_local_-start_idx_dest;
+  assert(howManyToCopy <= howManyToCopySrc);
```

Comment:
Do you think the assertion should be the other way around?
```suggestion
  int howManyToCopy = this->n_local_ - start_idx_dest;
  const int howManyToCopySrc = v.n_local_-start_idx_dest;
  assert(howManyToCopySrc <= howManyToCopy);
```
I would think that the number of source elements should not exceed the number of destination elements. Otherwise, one may end up writing passed the end of the array, which can lead to undefined behavior. I don't see anything in the code what would prevent this from happening.